Morphing (#63)
bmwcarit/ramses-composer-docs: doc/advanced/morphing/README.md

For further details of morphing in glTF see the [glTF morphing tutorial](https://github.com/KhronosGroup/glTF-Tutorials/blob/master/gltfTutorial/gltfTutorial_018_MorphTargets.md).

Note that morphing including animations of the morph weights is only supported starting at feature level 4.

## Import from glTF File

As with skinning the first step to setup morphing is the import from the glTF file. In this case no additional objects specific to morphing are present in the glTF import dialog although the animation contained in the example file targets the morph weights for the mesh as discussed below.

![](./docs/gltf-import.png)

Looking at the imported Mesh object we see the additional morph target attributes:

![](./docs/mesh-property-browser.png)

In this examples there are two morph targets, and the morph targets have both position and normal attributes. The attributes for morphing follow a simple naming scheme with a postfix `_Morph_N` attached to the base attribute name where `N`=0,1,2,... is the index of the morph target.

## Setting up the shader

> To make use the additional morph target vertex attributes a specialized shader needs to be supplied. A template shader is included with the RamsesComposer distribution. The vertex shader is reproduced below.

Reviewer: ```suggestion
To make use the additional morph target vertex attributes a specialized shader needs to be supplied. A template shader is included with the RamsesComposer distribution. For example, the vertex shader can look like this:
```
